custom_components/detailed_hello_world/README.md

```diff
@@ -0,0 +1,11 @@
+# More Comprehensive async Integration Example
+
+This example aims to show the best practice for a more complete integration.
+
+It is based on a _cover_, emulating battery operated roller blinds.
+
+The example includes extensive comments (that should be removed if making a true integration), to guide you through what each field and property is for. It includes 2 sensors tied to the primary cover device.
+
+It's all implemented using a _push_ model in _async_.
```

Comment:
I think that this example should be a "level 2" example, with an easier example being similar to this, but being based on polling an API and using the [DataUpdateCoordinator](https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities) as that will be probably the way to go for 75% of integrations.

Reply:
I'll give that a go (probably in a few days), and put in another PR.